services/brig/test/integration/API/User/Client.hs

```diff
@@ -518,11 +529,38 @@ testUpdateClient opts brig = do
     )
     !!! const 200
     === statusCode
+
   -- check if label is still present
   getClient brig uid (clientId c) !!! do
     const 200 === statusCode
     const (Just "label") === (clientLabel <=< responseJsonMaybe)
 
+  -- update supported client features
+  let checkUpdate :: HasCallStack => Maybe [SupportedClientFeature] -> Bool -> [SupportedClientFeature] -> Http ()
+      checkUpdate featuresIn respStatusOk featuresOut = do
+        let update'' = UpdateClient [] Nothing Nothing featuresIn
+        put
+          ( brig
+              . paths ["clients", toByteString' (clientId c)]
+              . zUser uid
+              . contentJson
+              . body (RequestBodyLBS $ encode update'')
+          )
+          !!! if respStatusOk
+            then do
+              const 200 === statusCode
+            else do
+              const 409 === statusCode
+              const (Just "client-feature-cannot-be-removed") === fmap Error.label . responseJsonMaybe
+
+        getClientSupportedFeatures brig uid (clientId c) !!! do
+          const 200 === statusCode
+          const (Just (SupportedClientFeatureList featuresOut)) === responseJsonMaybe
+
+  checkUpdate (Just [ClientSupportsLegalholdImplicitConsent]) True [ClientSupportsLegalholdImplicitConsent]
+  checkUpdate Nothing True [ClientSupportsLegalholdImplicitConsent]
+  checkUpdate (Just []) False [ClientSupportsLegalholdImplicitConsent]
+
```

Comment:
Can you test that updating the client with only the supportedClientFeature doesn't break the rest (i.e. other parts of the client entry are not erased) ?

Reply:
b9469683055876b3927c255cd1098a06202ebd8b